src/pybind/mgr/cephadm/module.py

```diff
@@ -917,6 +935,129 @@ def _set_extra_ceph_conf(self, inbuf=None) -> HandleCommandResult:
+        self.set_store(kv_option, value)
+
+    def _get_exporter_option(self, option: str) -> Union[str, None]:
+        kv_option = f'exporter_{option}'
+        return self.get_store(kv_option)
+
+    @orchestrator._cli_write_command(
+        prefix='cephadm generate-exporter-config',
+        desc='Generate default SSL crt/key and token for cephadm exporter daemons')
+    @service_inactive('cephadm-exporter')
+    def _generate_exporter_config(self):
+        # if self._spec_exists():
+        #     return 1, "", f"cephadm-exporter is already deployed. Changing configuration against an active service is not permitted."
+
+        crt, key = self._generate_exporter_ssl()
+        token = self._generate_exporter_token()
+        self._set_exporter_option('crt', crt)
+        self._set_exporter_option('key', key)
+        self._set_exporter_option('token', token)
+
+        self._set_exporter_option('port', CEPHADM_EXPORTER_PORT)
+        self._set_exporter_option('enabled', 'true')
+        self.log.info('Default SSL settings created for cephadm exporter(s)')
+        return 0, "", ""
+
+    def _generate_exporter_ssl(self):
+        return create_self_signed_cert(dname={"O": "Ceph", "OU": "cephadm-exporter"})
+
+    def _generate_exporter_token(self):
+        return secrets.token_hex(32)
+
+    @orchestrator._cli_write_command(
+        prefix='cephadm clear-exporter-config',
+        desc='Clear the SSL configuration used by cephadm exporter daemons')
+    @service_inactive('cephadm-exporter')
+    def _clear_exporter_config(self):
+        self._clear_exporter_config_settings()
+        self.log.info('Cleared cephadm exporter configuration')
+        return 0, "", ""
+
+    def _clear_exporter_config_settings(self) -> None:
+        self._set_exporter_option('crt', None)
+        self._set_exporter_option('key', None)
+        self._set_exporter_option('token', None)
+        self._set_exporter_option('port', None)
+        self._set_exporter_option('enabled', None)
+
+    @orchestrator._cli_write_command(
+        prefix='cephadm set-exporter-tls',
+        desc='Set custom cephadm-exporter TLS configuration from a json file (-i <file>). JSON must contain crt and key')
+    @service_inactive('cephadm-exporter')
+    def _set_exporter_tls(self, inbuf=None):
+        # if self._spec_exists():
+        #     return 1, "", f"cephadm-exporter is already deployed. Changing configuration against an active service is not permitted."
+
+        if not inbuf:
+            return 1, "", "TLS configuration has not been provided (-i <filename>)"
+
+        try:
+            cfg = json.loads(inbuf)
+        except ValueError:
+            return 1, "", "Invalid JSON file, parsing error"
+
+        if not all(k in cfg for k in ['crt', 'key']):
+            return 1, "", "JSON file must contain crt and key"
+
+        try:
+            verify_tls(cfg['crt'], cfg['key'])
+        except ServerConfigException as e:
+            return 1, "", str(e)
```

Comment:
can you please move this into a 

```python
class CepadmExporterConfig:
    def validate_certs(..): ...
```

mainly for keeping the code structure maintainable.

Reply:
ack